src/django_idom/components.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+from django.contrib.staticfiles.finders import find
+from idom import component, html
+
+from django_idom.config import IDOM_CACHE
+
+
+@component
+def static_css(static_path: str):
+    """Fetches a CSS static file for use within IDOM. This allows for deferred CSS loading."""
+    # Try to find the file within Django's static files
+    abs_path = find(static_path)
+    if not abs_path:
+        raise FileNotFoundError(
+            f"Could not find static file {static_path} within Django's static files."
+        )
+
+    # Fetch the file from cache, if available
+    last_modified_time = os.stat(abs_path).st_mtime
+    cache_key = f"django_idom:css_contents:{static_path}"
+    file_contents = IDOM_CACHE.get(cache_key, version=last_modified_time)
+    if file_contents is None:
+        with open(abs_path, encoding="utf-8") as static_file:
+            file_contents = static_file.read()
+        IDOM_CACHE.delete(cache_key)
+        IDOM_CACHE.set(
+            cache_key, file_contents, timeout=None, version=last_modified_time
+        )
```

Comment:
If it weren't for the fact that IDOM is currently using `is` for string and float comparisons you could do this pretty cleanly with `use_memo`:

```python
from pathlib import Path

def static_css(static_path: str):
    abs_path = find(static_path)
    if not abs_path:
        raise FileNotFoundError(
            f"Could not find static file {static_path} within Django's static files."
        )

    return html.style(
        use_memo(
            lambda: Path(abs_path).read_text(encoding="utf-8"),
            dependencies=[abs_path, os.stat(abs_path).st_mtime],
        )
    )
```

I've already been thinking about special casing comparisons for ints, floats, strings etc to use `==` instead of `is` (which is basically how the `===` operator works in JS from what I can tell) and this example makes a good case for making that change.

Reply:
https://github.com/idom-team/idom/issues/773

Reply:
@rmorshea `use_memo` is not multiprocessing compatible, while `IDOM_CACHE` is.